### PR TITLE
fix: scale image input with NCNN

### PR DESCRIPTION
**Pull request by beniz**

### CODE REVIEW

File: src/backends/ncnn/ncnninputconns.h [hunk 1]
```diff
@@ -113,6 +113,20 @@ namespace dd
...
+            {
+              std::vector<float> vscale = { norm };
+              _in.substract_mean_normalize(0, vscale.data());
+            }
+          else
+            {
+              std::vector<float> vscale = { norm, norm, norm };
+              _in.substract_mean_normalize(0, vscale.data());
+            }
+        }
```
Comment (BynaryCobweb) [inline src/backends/ncnn/ncnninputconns.h:129]: In torch backend, scale is applied before std and mean. Should this be the preferred behavior ? I feel like if the user provides both scale and mean / std at the same time, and they are not supposed to, an exception should be thrown.

File: tests/ut-ncnnapi.cc [hunk 2]
```diff
@@ -85,4 +85,20 @@ TEST(ncnnapi, service_predict)
...
+        + incept_repo + "face.jpg\"]}";
+  joutstr = japi.jrender(japi.service_predict(jpredictstr));
+  std::cout << "joutstr=" << joutstr << std::endl;
+  jd.Parse<rapidjson::kParseNanAndInfFlag>(joutstr.c_str());
+  ASSERT_TRUE(!jd.HasParseError());
+  ASSERT_EQ(200, jd["status"]["code"]);
+  ASSERT_TRUE(jd["body"]["predictions"].IsArray());
+  cl1 = jd["body"]["predictions"][0]["classes"][0]["cat"].GetString();
+  ASSERT_TRUE(jd["body"]["predictions"][0]["classes"][0]["prob"].GetDouble()
+              > 0.4);
```
Comment (BynaryCobweb) [inline tests/ut-ncnnapi.cc:103]: This test checks that there are no errors when providing scale, but offer no guarantee that the scale is correctly applied

File: src/backends/ncnn/ncnninputconns.h (hunk 1 above)
Comment (beniz) [reply src/backends/ncnn/ncnninputconns.h:129]: Yes, raising the exception should go into another PR I believe.

File: tests/ut-ncnnapi.cc (hunk 2 above)
Comment (beniz) [reply tests/ut-ncnnapi.cc:103]: Yes, we don't have models like this yet, at least it goes through the code, that uses the same NCNN mat function as `mean` and `std`.

**Review by fantes — APPROVED** [review]

File: tests/ut-ncnnapi.cc (hunk 2 above)
Comment (BynaryCobweb) [reply tests/ut-ncnnapi.cc:103]: I can try to export a pytorch model to ncnn... Maybe in the future :)

**Review by BynaryCobweb — APPROVED** [review]

**Pull request merged**